File: 67-add-binary/add-binary.java
class Solution {
    public String addBinary(String a, String b) {
        char []c1=a.toCharArray();
        char []c2=b.toCharArray();
        StringBuilder answer = new StringBuilder();
        int i=c1.length-1 ,j=c2.length-1;
        int carry=0;
        while(i>=0 ||j>=0|| (carry>0))
        {
            int val1=(i>=0)?Character.getNumericValue(c1[i]):0;
            int val2=(j>=0)?Character.getNumericValue(c2[j]):0;
            int ans = (val1+val2+carry);
            answer.append(ans%2);
            carry= (ans)/2;
            i--;j--;
        }
        answer.reverse();
        return answer.toString();
    }
}

File: 67-add-binary/add-binary.cs
public class Solution {
    public string AddBinary(string a, string b) {
        int maximumValue = Math.Max(a.Length,b.Length);
        string ans="";
        char carry='0';
        int firstWord = a.Length-1;
        int secondWord = b.Length-1;
        for(int ind =0;ind<maximumValue;ind++){
            int val1=0;
            int val2=0;
            if(firstWord>=0){
                val1=a[firstWord--]-'0';
            }if(secondWord>=0){
                val2 =b[secondWord--]-'0';
            }
            int sum = val1+val2+(carry-'0');
           ans+=(char)((sum%2)+'0');
           carry=(char)((sum/2)+'0');
        }
    if(carry=='1'){
        ans+='1';
    }
        
        return new string(ans.Reverse().ToArray());
    }
}